modules/account-lib/src/coin/dot/keyPair.ts

```diff
@@ -0,0 +1,74 @@
+import { Keyring } from '@polkadot/keyring';
+import { createPair } from '@polkadot/keyring/pair';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { Ed25519KeyPair } from '../baseCoin';
+import { AddressFormat } from '../baseCoin/enum';
+import { DefaultKeys, KeyPairOptions } from '../baseCoin/iface';
+import utils from './utils';
+
+const TYPE = 'ed25519';
+const keyring = new Keyring({ type: TYPE });
+
+export class KeyPair extends Ed25519KeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key, or a public key
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+  }
+
+  /**
+   * Helper function to create the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   * @see https://polkadot.js.org/docs/api/start/keyring
+   */
+  protected createPolkadotPair(): KeyringPair {
+    const secretKey = this.keyPair.prv ? new Uint8Array(Buffer.from(this.keyPair.prv, 'hex')) : new Uint8Array();
+    const publicKey = new Uint8Array(Buffer.from(this.keyPair.pub, 'hex'));
+    return createPair({ toSS58: keyring.encodeAddress, type: TYPE }, { secretKey, publicKey });
+  }
+
+  /** @inheritdoc */
+  getAddress(format?: AddressFormat): string {
+    return this.createPolkadotPair().address;
+  }
+
+  /**
+   * Getting the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   */
+  getSigningKeyPair(): KeyringPair {
```

Comment:
What's the difference between the signing key and `his.keyPair.prv;`?

Reply:
Returns an instance of keyring pair which has other methods and data based of off the private and public key